.vscode/c_cpp_properties.json

```diff
@@ -0,0 +1,32 @@
+{
+    "configurations": [
+        {
+            "name": "Mac",
+            "includePath": [
+                "${workspaceFolder}/common/**",
```

Comment:
Do we need these? Aren't these fed to us by cmake?

Reply:
I actually don't know - I am a real newbie in C++ and just get started with CMake etc - I just added the distinct paths which had C++ files in it.